### DISCUSSION TITLE
layer 2 broadcast across networks via IPsec tunnels

### DISCUSSION THREAD

**Discussion by TomHsiung** [strongswan/strongswan#267] [Q&A]
Hello,

I wonder if functions that rely on LAN broadcasting (DHCP, AirPrint, ARP, AirPlay, etc.) could be available by strongSwan IPsec tunnels? You know, today I was outside and had to print a document. I intended to print it via my iPhone by AirPlay. However, my printer was not available even I connected my home network via strongSwan.

PS: it seems that iPhone does not support CUPS network printing, which Mac indeed supports.

Thank you.

**Comment by Thermi** [comment]
Hi,

You need to use the forecast plugin to forward multicast and broadcast packets over the CHILD_SAs. For that, the traffic selectors of the CHILD_SA need to cover the broadcast and multicast addresses, too. Take a look at the [wiki](https://wiki.strongswan.org/projects/strongswan/wiki/Forecast) for that.

**Reply by TomHsiung** [reply]
Looks like the forecast plugin is under experiment. I cannot install it. Could you please tell me how? Thank you.

**Reply by Thermi** [reply]
Why can you not install it?

**Reply by TomHsiung** [reply]
I don't know how to install it. I have installed

`libcharon-extauth-plugins`
`libcharon-extra-plugins`
`libcharon-standard-plugins`

However, there is no forecast plugin conf file under directory `/etc/strongswan.d/charon/`
